pkg/authenticator/jwt/authenticator.go

```diff
@@ -0,0 +1,130 @@
+package jwt
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+	"go.opentelemetry.io/otel"
+	"net/http"
+	"os"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/access_token"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator/common"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/utils"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/trace"
+)
+
+// Authenticator contains the configuration and client
+// for the authentication connection to Conjur
+type Authenticator struct {
+	client      *http.Client
+	privateKey  *rsa.PrivateKey
+	AccessToken access_token.AccessToken
+	Config      *Config
+	PublicCert  *x509.Certificate
```

Comment:
Any reason for making these public ? I also realised that the Authenticator interface should expose `AccessToken` through some getter. I think we spoke about having Authenticate and GetAccessToken on the call.

Reply:
Changes to non public.

But don't see why we need getAccessToken didn't found usage of authn.AccessToken in secretless and in provider.
tell me if im wrong and i will add it back